java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -43,6 +43,8 @@
   int resultIndex;
   @Nullable
   Constraint resultConstraint;
+  @Nullable
+  String exceptionType;
```

Comment:
If I'm correct this information is sufficient to remove the boolean exception and transform its usage to a method call checking for nullness of this String ?